### PR TITLE
Comparator result negation fix

### PR DESCRIPTION
**Pull request by DanielBenniah**
https://github.com/Azure/azure-sdk-for-java/issues/9109

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/orderbyquery/OrderbyRowComparer.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public int compare(OrderByRowResult<T> r1, OrderByRowResult<T> r2) {
                     case Ascending:
                         return cmp;
                     case Descending:
-                        return -cmp;
+                        return ItemComparator.getInstance().compare(result2.get(i).getItem(), result1.get(i).getItem());
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/orderbyquery/OrderbyRowComparer.java]: This recomputes the cmp, instead you can use signum

Comment (DanielBenniah) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/orderbyquery/OrderbyRowComparer.java]: Yes, I was debating the same if I should recompute or use +1/-1.

Using signum is a good idea. I've made the change. I've changed the Ascending case as well to keep the result consistent. Please let me know if the change looks good.

**Comment by DanielBenniah** [comment]
@moderakh How do I re-run the ci?

**Comment by DanielBenniah** [comment]
@moderakh All checks have passed! Thank you!

**Comment by kushagraThapar** [comment]
> @moderakh All checks have passed! Thank you!

Looks good, I will get this in now.

**Pull request merged**